src/sentry/static/sentry/app/views/dashboardsV2/widgetQueries.tsx

```diff
@@ -0,0 +1,173 @@
+      // replied with a map like: {[titleString: string]: EventsStats}
+      const result: Series[] = Object.keys(series)
+        .map((seriesName: string): [number, Series] => {
+          const prefixedName = seriesNamePrefix
+            ? `${seriesNamePrefix} : ${seriesName}`
+            : seriesName;
+          const seriesData: EventsStats = series[seriesName];
+          const transformed = transformSeries(seriesData, prefixedName);
+          return [seriesData.order || 0, transformed];
+        })
+        .sort((a, b) => a[0] - b[0])
+        .map(item => item[1]);
+
+      output = output.concat(result);
+    } else {
+      const field = widget.queries[i].fields[0];
+      const prefixedName = seriesNamePrefix ? `${seriesNamePrefix} : ${field}` : field;
+      const transformed = transformSeries(series, prefixedName);
+      output.push(transformed);
+    }
+  }
+  return output;
+}
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  widget: Widget;
+  selection: GlobalSelection;
+  children: (props: Pick<State, 'loading' | 'error' | 'results'>) => React.ReactNode;
+};
+
+type State = {
+  error: boolean;
+  loading: boolean;
+  results: Series[];
+};
+
+class WidgetQueries extends React.Component<Props, State> {
+  state: State = {
+    loading: true,
+    error: false,
+    results: [],
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    const {selection, widget} = this.props;
+    if (
+      !isEqual(widget.queries, prevProps.widget.queries) ||
+      !isEqual(selection, prevProps.selection)
+    ) {
+      this.fetchData();
+    }
+  }
+
+  async fetchData() {
+    const {selection, api, organization, widget} = this.props;
+
+    this.setState({loading: true});
+
+    const statsPeriod = selection.datetime.period;
+    const {start, end} = selection.datetime;
+    const {projects, environments} = selection;
+    const interval = getWidgetInterval(widget.interval, {
+      start,
+      end,
+      period: statsPeriod,
+    });
+
+    const promises: Promise<EventsStats | MultiSeriesEventsStats>[] = widget.queries.map(
+      query => {
+        // TODO(mark) adapt this based on the type of widget being built.
+        // Table and stats results will need to do a different request.
+        const requestData = {
+          organization,
+          interval,
+          start,
+          end,
+          project: projects,
+          environment: environments,
+          period: statsPeriod,
+          query: query.conditions,
+          yAxis: query.fields,
+          includePrevious: false,
+        };
+        return doEventsRequest(api, requestData);
+      }
+    );
+
+    try {
+      // Use Promise.all() to get this working. Ideally as results become available
+      // they are added to the component state and incrementally exposed to children.
+      const rawResults = await Promise.all(promises);
```

Comment:
If you want it to incrementally load in results, you can forEach the promises with an async function and await each to call this setState. 

Something like this:
```js
let incrementalResults = [];
promises.forEach(async (p) => {
  const result = await p;
  incrementalResults.push(result);
  this.setState({
    results: transformResults(widget, incrementalResults),
    loading: false,
    error: false,
  });
});
```

The only caveat is it might cause a bunch of render updates in close succession, might be minor though.

Reply:
Thanks! I'll incorporate this in a followup set of changes as there is still lots to do on this modal.